File: manual/data/python/src/function_request_get.py
import requests


class RequestHandler:

    def handle_request(self, url: str) -> str:
        try:
            response = requests.get(url, timeout=5, allow_redirects=False)
        except requests.exceptions.Timeout:
            return "timeout"

        if response.status_code == 200:
            try:
                if not response.content.strip():
                    return "non-json"
                response.json()
                return "success"
            except requests.exceptions.JSONDecodeError:
                return "non-json"
        elif 300 <= response.status_code < 400:
            return f"redirect {response.status_code}"
        else:
            return f"error {response.status_code}"

File: manual/code/generate_strategies/IR(CoT)/java/java/function_request_get.java
package com.example;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.ResponseBody;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.concurrent.TimeUnit;

public class RequestHandler {
    public String handle_request(String url) {
        class RequestHandler {
            private final OkHttpClient client;
            private final ObjectMapper objectMapper;

            RequestHandler() {
                this.client = new OkHttpClient.Builder()
                        .connectTimeout(5, TimeUnit.SECONDS)
                        .readTimeout(5, TimeUnit.SECONDS)
                        .followRedirects(false)
                        .build();
                this.objectMapper = new ObjectMapper();
            }

            String processRequest(String url) {
                Request request = new Request.Builder()
                        .url(url)
                        .build();

                try (Response response = client.newCall(request).execute()) {
                    int statusCode = response.code();

                    if (statusCode == 200) {
                        ResponseBody body = response.body();
                        if (body == null) {
                            return "non-json";
                        }

                        
                        String responseStr = body.string();

                        if (responseStr.trim().isEmpty()) {
                            return "non-json";
                        }

                        try {
                            
                            objectMapper.readTree(responseStr);
                            return "success";
                        } catch (Exception e) {
                            return "non-json";
                        }
                    } else if (statusCode >= 300 && statusCode < 400) {
                        return "redirect " + statusCode;
                    } else {
                        return "error " + statusCode;
                    }
                } catch (Exception e) {
                    return "timeout";
                }
            }
        }

        RequestHandler handler = new RequestHandler();
        return handler.processRequest(url);
    }
}